Documentation/ApiOverview/Rte/InTheFrontend/Introduction.rst

```diff
@@ -0,0 +1,116 @@
+about adding an RTE on frontend pages.
+
+.. figure:: ../Images/RteBackend.png
+   :alt: A RTE in the TYPO3 BE
+
+   The rtehtmlarea RTE activated in the TYPO3 backend
+
+.. tip::
+
+   This description about CKEditor includes examples and common challenges for
+   the frontend.
+   
+   You can use other editors with TYPO3 and some points, like
+   handling of data on the server, are independent of the distinct editor
+   in the frontend.
+
+.. note::
+
+   If you feel able to provide a description about using other editors
+   with TYPO3, please add to this documentation.
+
+The following list describes features and corresponding implementation
+effort ordered from simple to complex. 
+
+The optional features
+=====================
+
+#. Simple text formatting can be achieved using well-known buttons.
+   This solution is used to format text (bold, italic, underlined, ...),
+   create lists or tables, etc..
+   These options are predefined in the RTE and
+   and wrap selected content in html-tags, by default without any attributes
+   like `id` or `class` for example.
+
+#. Advanced text-formatting can be achieved with predefined blocks and
+   according style. Those blocks wrap selected content in html-elements 
+   with CSS-classes that can be styled in a stylesheet. The formats have
+   to be defined by names, short description and according styling.
+   CKEditor offers a dropdown button for those block-styles.
+   
+   Editing the Source could allow the user optionally to add special
+   HTML-elements or attributes like `id`, `class` or more.
+
+#. It might be desired to allow users to upload files like images,
+   PDF-documents or other multi-media-content. Images and perhaps some
+   other file-types could be displayed in the content, further file-types
+   could be linked.
+
+#. Editing data in the frontend requires that applying forms are pre-filled
+   with existing data. This might require some considerations concerning
+   multiple aspects.
+
+#. Links might be choosen out of the existing pages of the website, those
+   links can be added as internal instead of external links but require
+   a visual and functional option to select from existing pages.
+   
+   *This option requires an AJAX-connection to interact with TYPO3.*
+
+#. For special websites like intranets it might be desired additionally
+   to not only allow the upload of media but also to choose media out of
+   those that exist already in a public directory on the server.
+   
+   *This option requires an AJAX-connection to interact with TYPO3.*
+
+.. important::
+
+   User input in the frontend always represents a general security risk.
+
+   **A Rich Text Editor might reduce this awareness** because it looks like a
+   professional solution but in fact might increase vulnerability if features
+   are enabled or included without proper handling on the server.
+
+   It's advised to allow only required input and to disallow any further
```

Comment:
link to xss attack descriptions in security chapter? https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/Security/TypesOfThreats/Index.html#cross-site-scripting-xss

Reply:
yes!